### PR TITLE
flask_2 homework

### PR DESCRIPTION
**Pull request by Odin2345**

### CODE REVIEW

File: flask_lesson_intro/main.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+
+@app.route('/headphones')
+def get_headphones_page():
+    return render_template("product.html", data=get_data(), title="Headphones")
+
+
+@app.route('/ipod')
+def get_ipod_page():
+    return render_template("product.html", data=get_data(), title="iPod")
```
Comment (dimigor) [inline flask_lesson_intro/main.py]: you do not need all this routes with template, try to make universal route

File: flask_lesson_intro/templates/author.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{% extends "base.html" %}
+{% block info %}
+<h3 style="padding: 25px"><a href="https://github.com/Odin2345">Serhii Liakh</a></h3>
```
Comment (dimigor) [inline flask_lesson_intro/templates/author.html]: missed tabulation

File: flask_lesson_intro/utils.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import json
+
+
+def get_data():
+    with open("data.json") as file:
+        return json.load(file)
```
Comment (dimigor) [inline flask_lesson_intro/utils.py]: what if the file will not found or contains not valid data

**Review by dimigor — CHANGES_REQUESTED** [review]

File: flask_lesson_intro/main.py (hunk 1 above)
Comment (Odin2345) [reply flask_lesson_intro/main.py]: fixed

File: flask_lesson_intro/templates/author.html (hunk 2 above)
Comment (Odin2345) [reply flask_lesson_intro/templates/author.html]: fixed

File: flask_lesson_intro/utils.py (hunk 3 above)
Comment (Odin2345) [reply flask_lesson_intro/utils.py]: FileNotFoundError: [Errno 2] No such file or directory: 'data.json' 

json.decoder.JSONDecodeError: Expecting ',' delimiter: line 14 column 3 (char 2063)

– we can see one of these errors )

Comment (dimigor) [reply flask_lesson_intro/utils.py]: Great, maybe you should catch them?

File: flask_lesson_intro/templates/product.html [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{% extends "base.html" %}
+{% block info %}
+<div align="center">
+    <div class="card">
+        {% for elem in data %}
+        {% if elem['title'] == product %}
```
Comment (dimigor) [inline flask_lesson_intro/templates/product.html]: maybe you should do this logic at route?

File: flask_lesson_intro/utils.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+import logging
+
+
+def get_data():
+    try:
+        with open("data.json") as file:
+            return json.load(file)
+    except ValueError:
+        logging.critical("File 'data.json' contains not valid data")
+    except FileNotFoundError:
```
Comment (dimigor) [inline flask_lesson_intro/utils.py:11]: great!)

**Review by dimigor — CHANGES_REQUESTED** [review]

**Review by dimigor — APPROVED** [review]

**Pull request merged**